### PR TITLE
gh-67693: Fix urlunparse() and urlunsplit() for URIs with path starting with multiple slashes and no authority

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Based on #15297.

Co-authored-by: Ashwin Ramaswami <aramaswamis@gmail.com>

* Issue: gh-78457

* Issue: gh-67693

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
It fixes also more serious security issue #67693.

File: Misc/NEWS.d/next/Library/2019-08-27-01-16-50.gh-issue-67693.4NIAiy.rst
Comment (vadmium) [inline Misc/NEWS.d/next/Library/2019-08-27-01-16-50.gh-issue-67693.4NIAiy.rst:1]: Are there meant to be two _NEWS.d_ files?

**Review by vadmium — COMMENTED** [review]
The code changes and tests look good to me.

File: Misc/NEWS.d/next/Library/2019-08-27-01-16-50.gh-issue-67693.4NIAiy.rst
Comment (serhiy-storchaka) [reply Misc/NEWS.d/next/Library/2019-08-27-01-16-50.gh-issue-67693.4NIAiy.rst:1]: Good catch.

File: Lib/test/test_urlparse.py [hunk 1]
```diff
@@ -175,9 +175,39 @@ def test_qs(self):
...
+            ('path/to/file',
+             ('', '', 'path/to/file', '', '', ''),
+             ('', '', 'path/to/file', '', '')),
+            ('/path/to/file',
+             ('', '', '/path/to/file', '', '', ''),
+             ('', '', '/path/to/file', '', '')),
+            ('//path/to/file',
+             ('', 'path', '/to/file', '', '', ''),
+             ('', 'path', '/to/file', '', '')),
+            ('////path/to/file',
```
Comment (serhiy-storchaka) [inline Lib/test/test_urlparse.py:189]: This case was broken.

File: Lib/test/test_urlparse.py [hunk 2]
```diff
@@ -175,9 +175,39 @@ def test_qs(self):
...
+            ('scheme:path/to/file',
+             ('scheme', '', 'path/to/file', '', '', ''),
+             ('scheme', '', 'path/to/file', '', '')),
+            ('scheme:/path/to/file',
+             ('scheme', '', '/path/to/file', '', '', ''),
+             ('scheme', '', '/path/to/file', '', '')),
+            ('scheme://path/to/file',
+             ('scheme', 'path', '/to/file', '', '', ''),
+             ('scheme', 'path', '/to/file', '', '')),
+            ('scheme:////path/to/file',
```
Comment (serhiy-storchaka) [inline Lib/test/test_urlparse.py:201]: This case was broken.

File: Lib/test/test_urlparse.py [hunk 3]
```diff
@@ -175,9 +175,39 @@ def test_qs(self):
...
+             ('scheme', '', '//path/to/file', '', '', ''),
+             ('scheme', '', '//path/to/file', '', '')),
             ('file:///tmp/junk.txt',
              ('file', '', '/tmp/junk.txt', '', '', ''),
              ('file', '', '/tmp/junk.txt', '', '')),
+            ('file:////tmp/junk.txt',
```
Comment (serhiy-storchaka) [inline Lib/test/test_urlparse.py:207]: This case was broken.

File: Lib/test/test_urlparse.py [hunk 4]
```diff
@@ -175,9 +175,39 @@ def test_qs(self):
...
+             ('file', '', '//tmp/junk.txt', '', '', ''),
+             ('file', '', '//tmp/junk.txt', '', '')),
+            ('file://///tmp/junk.txt',
```
Comment (serhiy-storchaka) [inline Lib/test/test_urlparse.py:210]: This case was broken.

**Review by sethmlarson — APPROVED** [review]
LGTM!

**Pull request merged**